Mathlib/RingTheory/Regular/Category.lean

```diff
@@ -0,0 +1,52 @@
+/-
+Copyright (c) 2025 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jingting Wang, Wanyi He, Nailin Guan
+-/
+import Mathlib.Algebra.Homology.ShortComplex.ModuleCat
+import Mathlib.RingTheory.QuotSMulTop
+
+/-!
+# Categorical constructions for `IsSMulRegular`
+-/
+
+universe u v w
+
+variable {R : Type u} [CommRing R] (M : ModuleCat.{v} R)
+
+open CategoryTheory Ideal Pointwise
+
+/-- The short complex `M → M → M⧸xM` given by an element `x : R`. -/
+@[simps]
+def smulShortComplex (r : R) :
```

Comment:
As this involves objects in `ModuleCat`, I would suggest placing almost everything apart from `IsSMulRegular.smulShortComplex_shortExact` in the `ModuleCat` namespace.